MapboxCoreNavigation/MBNavigator.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+import MapboxNavigationNative
+import CoreLocation
+
+extension MBFixLocation {
+    
+    convenience init(_ location: CLLocation) {
+        self.init(location: location.coordinate,
+                  time: location.timestamp,
+                  speed: location.speed as NSNumber,
+                  bearing: location.course as NSNumber,
+                  altitude: location.altitude as NSNumber,
+                  accuracyHorizontal: location.horizontalAccuracy as NSNumber,
+                  provider: nil)
+    }
+}
+
+extension MBRouteState: CustomStringConvertible {
+    
+    public var description: String {
+        switch self {
+        case .invalid:
+            return "invalid"
+        case .initialized:
+            return "initialized"
+        case .tracking:
+            return "tracking"
+        case .complete:
+            return "complete"
+        case .offRoute:
+            return "offRoute"
+        case .stale:
+            return "stale"
+        }
+    }
+}
+
+extension MBNavigationStatus {
+    
+    var isFirstInstructionOnFirstStep: Bool {
```

Comment:
`isAtFirstInstructionOfLeg`? Does this logic need to account for multi-leg routes?

Reply:
This is actually dead code so I removed it. nav-native is handling this now.